### PR TITLE
Tasmota driver refactor

### PR DESCRIPTION
**Pull request by hectorespert**
Rewrite parts of the Tasmota driver and fix minor style comments from the first PR: https://github.com/reef-pi/drivers/pull/75

### CODE REVIEW

File: tasmota/http.go [hunk 1]
```diff
@@ -51,72 +52,86 @@ func (m *httpDriver) PWMChannel(_ int) (hal.PWMChannel, error) {
...
+func (m *httpDriver) doRequest(url string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
-		return false
+		return nil, err
 	}
 	c := http.Client{
 		Timeout: 5 * time.Second,
 	}
 	resp, err := c.Do(req)
```
Comment (ranjib) [inline tasmota/http.go:63]: just 
```go
return  c.Do(req)
```

**Comment by ranjib** [comment]
left some style nits, rest look good

File: tasmota/http.go (hunk 1 above)
Comment (hectorespert) [reply tasmota/http.go]: Changed

**Comment by ranjib** [comment]
Thank you so much :-) . I am relying on you to test it out for the time being. I'll see if I can find some time to flash my wifi outlet with tasmota. I think shelly works on tasmota

**Pull request merged**